homework.cls

```diff
@@ -50,9 +52,13 @@
 % default value and will raise an error if used before being set by their
 % corresponding defining macros.
 \def\@name{\@latex@error{No \noexpand\name given}\@ehc}
+\def\@studentid{\@latex@error{No \noexpand\studentid given}\@ehc}
+\def\@email{\@latex@error{No \noexpand\email given}\@ehc}
 \def\@course{\@latex@error{No \noexpand\course given}\@ehc}
+\def\@courseid{\@latex@error{No \noexpand\courseid given}\@ehc}
 \def\@term{\@latex@error{No \noexpand\term given}\@ehc}
 \def\@hwnum{\@latex@error{No \noexpand\hwnum given}\@ehc}
+\def\@duedate{\@latex@error{No \noexpand\duedate given}\@ehc}
```

Comment:
these new mandatory options break the existing `example.tex` and `template.tex` for me. Perhaps they should be optional? Or maybe the old options should be optional as well.

Reply:
This is one of the changes I didn't mean to propose. I'm using it, but they might be more specific than needed for most people?